### ISSUE TITLE
File transfer

### ISSUE THREAD

**Issue by andreskru** [emesene/emesene#102]
When i accept a transfer and it is complete, I cant found the file!!!
where is it?

running Fedora 13

**Comment by c10ud** [comment]
uhm, emesene-dir?

**Comment by DragaO** [comment]
i'm using fedora 13 too

i tried to send a file (for another cliente) and it's ok, but, i can't receive files.
i tried to accept the file but nothing has happened

**Comment by c10ud** [comment]
we need to discuss where to place the config value for received-file-transfers, and if it can be done globally

**Comment by hookdump** [comment]
Something like $HOME/Downloads or $HOME/Recieved would be nice. I mean, something more "global" and intuitive for users.

**Comment by marianoguerra** [comment]
I think we shuld use this list falling back to the one bellow if not set
1. user defined in config config.get("download_dir", None)
2. $XDG_DOWNLOAD_DIR
3. xdg-user-dir DOWNLOAD if the program is available
4. the value of XDG_DOWNLOAD_DIR in ~/.config/user-dirs.dirs if not sourced
5. $HOME/Downloads if exists (or maybe $(xdg-user-dir)/Downloads ?)
6. $HOME?

**Comment by c10ud** [comment]
uhm, i was thinking of e3/common but seems like i can't access configuration from there, right?

[ the ideal thing would be a simple call path = Module.Name.get_full_filetransfer_path(filename, sender) where sender is an email and filename, the filename..then the module takes care of all the fallback list and adds eventual needed subdirectories (configurable value "save files in directories per-sender-account") ]

**Comment by marianoguerra** [comment]
something like this seems clean enough and we separate the config part from the "look for the directory part"

```
download_path = config.get_or_set("download_folder", e3.places.download())
full_path = os.path.join(download_path, filename)
```

we shouldn't use the download path in to many places so it won't cause too much code.

I think the "directory per sender" is an overkill but it would be only an extra if there

```
download_path = config.get_or_set("download_folder", e3.places.download())
if config.get_or_set("download_folder_per_account", False):
    full_path = os.path.join(download_path, account, filename)
else:
    full_path = os.path.join(download_path, filename)
```

not too much code, easy to understand and separates the protocol part from the config part.

wadayathink?

I will start to work on this if you are ok with it

**Comment by c10ud** [comment]
Update on filetransfers. Closed by e722ee80cf9fc6794e7c93d629260e2d20d40ef0

Now filetransfers are saved in the right directories (No gui for configuration yet)
Small fixes in the whole filetransfer gui (it's still weak and buggy though)
Added Desktop module in gui/base by Paul Boddie paul@boddie.org.uk (for Desktop.open)
